cmd/agent-js-worker/main.go

```diff
@@ -500,99 +526,320 @@ func createVDRs(resolvers []string, trustblocDomain, trustblocResolver string) (
 	return VDRs, nil
 }
 
-func agentOpts(opts *agentStartOpts) ([]aries.Option, error) {
+func agentOpts(startOpts *agentStartOpts) ([]aries.Option, error) {
 	msgHandler := msghandler.NewRegistrar()
 
-	var options []aries.Option
-	options = append(options, aries.WithMessageServiceProvider(msgHandler))
+	var ariesOptions []aries.Option
+	ariesOptions = append(ariesOptions, aries.WithMessageServiceProvider(msgHandler))
 
-	if opts.TransportReturnRoute != "" {
-		options = append(options, aries.WithTransportReturnRoute(opts.TransportReturnRoute))
+	if startOpts.TransportReturnRoute != "" {
+		ariesOptions = append(ariesOptions, aries.WithTransportReturnRoute(startOpts.TransportReturnRoute))
 	}
 
-	store, err := createStore(opts)
+	var err error
+
+	ariesOptions, err = addStorageAndKMSOptionsAsNeeded(startOpts, ariesOptions)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create storage provider: %w", err)
+		return nil, fmt.Errorf("unexpected failure while adding storage and KMS options (as needed): %w", err)
 	}
 
-	VDRs, err := createVDRs(opts.HTTPResolvers, opts.BlocDomain, opts.TrustblocResolver)
+	VDRs, err := createVDRs(startOpts.HTTPResolvers, startOpts.BlocDomain, startOpts.TrustblocResolver)
 	if err != nil {
 		return nil, err
 	}
 
 	for i := range VDRs {
-		options = append(options, aries.WithVDR(VDRs[i]))
+		ariesOptions = append(ariesOptions, aries.WithVDR(VDRs[i]))
 	}
 
-	options = append(options, aries.WithStoreProvider(store))
-
-	for _, transport := range opts.OutboundTransport {
+	for _, transport := range startOpts.OutboundTransport {
 		switch transport {
 		case "http":
 			outbound, err := arieshttp.NewOutbound(arieshttp.WithOutboundHTTPClient(&http.Client{}))
 			if err != nil {
 				return nil, err
 			}
 
-			options = append(options, aries.WithOutboundTransports(outbound))
+			ariesOptions = append(ariesOptions, aries.WithOutboundTransports(outbound))
 		case "ws":
-			options = append(options, aries.WithOutboundTransports(ws.NewOutbound()))
+			ariesOptions = append(ariesOptions, aries.WithOutboundTransports(ws.NewOutbound()))
 		default:
 			return nil, fmt.Errorf("unsupported transport : %s", transport)
 		}
 	}
 
-	return options, nil
+	return ariesOptions, nil
 }
 
-func createStore(opts *agentStartOpts) (storage.Provider, error) {
+func addStorageAndKMSOptionsAsNeeded(startOpts *agentStartOpts,
```

Comment:
could have been called addOptions -- you really like long names

Reply:
Changed the name to `addStorageOptions`. I didn't use just `addOptions` since that function only adds a subset of the options.